middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -3775,48 +3797,36 @@ let close_program (type mode) ~(mode : mode Flambda_features.mode) ~big_endian
     | Some [approx] -> approx
     | _ -> Value_approximation.Value_unknown
   in
-  let acc, body =
-    bind_code_and_sets_of_closures (Acc.code acc)
-      (Acc.lifted_sets_of_closures acc)
-      acc body
-  in
   (* We must make sure there is always an outer [Let_symbol] binding so that
      lifted constants not in the scope of any other [Let_symbol] binding get put
      into the term and not dropped. Adding this extra binding, which will
      actually be removed by the simplifier, avoids a special case. *)
-  let acc =
+  let acc, body =
     match Acc.declared_symbols acc with
-    | _ :: _ -> acc
+    | _ :: _ -> acc, body
     | [] ->
       (* CR vlaviron/mshinwell: Maybe this could use an empty array.
          Furthermore, can this hack be removed? *)
-      let acc, (_sym : Symbol.t) =
-        register_const0 acc
-          (Static_const.block Tag.Scannable.zero Immutable Value_only [])
-          "first_const"
+      let acc, symbol = manufacture_symbol acc "first_const" in
```

Comment:
Please document that we cannot use `register_const` here (because we've already extracted the declared symbols from the accumulator, if I understand correctly).